packages/api/portfolioDrafts/application/createApplicationStep.ts

```diff
@@ -101,111 +74,18 @@ export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayPr
- * @returns a collection of application validation errors
- */
-export function performDataValidationOnEnvironment(environment: Environment): Array<ApplicationStepValidationErrors> {
-  const errors = Array<ApplicationStepValidationErrors>();
-  if (environment.name.length < 4 || environment.name.length > 100) {
-    errors.push({
-      environmentName: environment.name,
-      invalidParameterName: "name",
-      invalidParameterValue: environment.name,
-      validationMessage: ValidationMessage.INVALID_ENVIRONMENT_NAME,
-    });
-  }
-  const operatorErrors: ApplicationStepValidationErrors[] = environment.operators
-    .map(performDataValidationOnOperator)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-  return errors.concat(operatorErrors);
-}
-
-/**
- * Performs data validation on the given Operator object
- * @param operator the object containing data to validate
- * @returns a collection of operator validation errors
- */
-export function performDataValidationOnOperator(operator: Operator): Array<ApplicationStepValidationErrors> {
-  const errors = Array<ApplicationStepValidationErrors>();
-  if (operator.display_name.length < 1 || operator.display_name.length > 100) {
-    errors.push({
-      operatorDisplayName: operator.display_name,
-      invalidParameterName: "display_name",
-      invalidParameterValue: operator.display_name,
-      validationMessage: ValidationMessage.INVALID_OPERATOR_NAME,
-    });
-  }
-  // Using API Gateway "email" format to enforce the most common email rules
-  // and only ensuring the .mil top-level domain here.
-  // TODO: possibly add patterns into the internal API spec to help
-  const emailRegex = /.mil$/i;
-  if (!emailRegex.test(operator.email)) {
-    errors.push({
-      operatorEmail: operator.email,
-      invalidParameterName: "email",
-      invalidParameterValue: operator.email,
-      validationMessage: ValidationMessage.INVALID_OPERATOR_EMAIL,
-    });
-  }
-  return errors;
-}
-
-/**
- * Returns an error response containing 1) an error map containing the specified invalid properties, 2) an error code, 3) a message
- * @param invalidProperties object containing property names and their values which failed validation
- * @returns ValidationErrorResponse containing an error map, error code, and a message
- */
-export function createValidationErrorResponse(invalidProperties: Record<string, unknown>): ValidationErrorResponse {
-  if (Object.keys(invalidProperties).length === 0) {
-    throw Error("Parameter 'invalidProperties' must not be empty");
-  }
-  Object.keys(invalidProperties).forEach((key) => {
-    if (!key) throw Error("Parameter 'invalidProperties' must not have empty string as key");
-  });
-  return new ValidationErrorResponse("Invalid input", invalidProperties);
-}
+const wrappedApplicationStepSchema = {
+  type: "object",
+  required: ["body"],
+  properties: {
+    body: applicationStepSchema,
+  },
+};
```

Comment:
To avoid repetition, I wrote a function (utils/schemaWrapper.ts) to wrap the schema and save us 6 lines: 
```
export function wrapSchema(schema: Record<string, unknown>): Record<string, unknown> {
  const wrappedSchema = {
    type: "object",
    required: ["body"],
    properties: {
      body: schema,
    },
  };
  return wrappedSchema;
}
```
You would then call it with:
```
.use(validator({ inputSchema: wrapSchema(schema.ApplicationStep)}))
```

Reply:
Awesome! I was considering a func for this since it would be the same everywhere middy is being used.